libs/utils/tests/logger_json_test.rs

```diff
@@ -0,0 +1,47 @@
+use std::fs::{File, OpenOptions};
+use std::io::{self, BufRead};
+use tempfile::*;
+use tracing::*;
+use utils::logging::*;
+
+fn read_lines(file: File) -> io::Lines<io::BufReader<File>> {
+    io::BufReader::new(file).lines()
+}
+
+#[test]
+fn test_json_format_has_message_and_custom_field() {
+    // test env filter
+    std::env::set_var("RUST_LOG", "info");
+
+    let tmp_dir = TempDir::new().unwrap();
+    let log_file_path = tmp_dir.path().join("logfile");
+
+    let log_format = LogFormat::Json;
+    let daemonize = true; //So that log is written to file
+    let _log_file = init(&log_file_path, daemonize, log_format).unwrap();
+    assert!(log_file_path.exists());
+
+    let custom_field: &str = "hi";
+    trace!(custom = %custom_field, "test log message");
+    debug!(custom = %custom_field, "test log message");
+    info!(custom = %custom_field, "test log message");
+    warn!(custom = %custom_field, "test log message");
+    error!(custom = %custom_field, "test log message");
+
+    let log_file = OpenOptions::new().read(true).open(&log_file_path).unwrap();
+
+    let lines = read_lines(log_file);
+    for line in lines {
+        let content = line.unwrap();
+        let json = serde_json::from_str::<serde_json::Value>(&content);
+        assert!(json.is_ok());
+
+        let json_object = json.unwrap();
```

Comment:
Especially in tests, it's always better to just `unwrap` or use question-mark operator `?` to get rid of the Result.

Compare these two assertion "failures":

`assert!(json.is_ok());` gives you something like:

```
thread 'main' panicked at 'assertion failed: json.is_ok()', src/main.rs:3:5
```

Whereas `unwrap` would:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "foobar"', src/main.rs:3:22
```

Reply:
```suggestion
        let json_object = json.unwrap();
```

Perhaps this could be combined as well with the `from_str` call, since the intermediate value has no use.

Reply:
Yeah, could iondeed be a bit too verbose, I'll clean it up.